File: src/wrangle.py
# author: Elliott Ribner
# date: 2020-01-20

'''This script saves handles the cleaning and preprocessing of the dataset imported in dl_xls_to_csv.py.

Usage: src/wrangle.py --input=<input> --output=<output>

Options:
--input=<input>  Name of csv file to download, must be within the /data directory.
--output=<output>  Name of file to be saved, to be stored in /data directory. Include .csv filetype. 
'''

import pandas as pd
from docopt import docopt

opt = docopt(__doc__)


def main(input, output):
    # download xls to pandas dataframe
    df = pd.read_csv(f"./data/{input}", index_col=1, skiprows=1)
    df = df.drop(columns=['0'])
    df = df.rename(
        columns={'default payment next month': 'DEFAULT_NEXT_MONTH', 'PAY_0': 'PAY_1'})
    df.to_csv(r"./data/%s" % (output))


if __name__ == "__main__":
    main(input=opt["--input"], output=opt["--output"])

File: src/wrangle.R
# author: Elliott Ribner
# date: 2020-01-20


"This script saves handles the cleaning and preprocessing of the dataset imported in dl_xls_to_csv.py.

Usage: src/wrangle.R <input> <output>

Options:
<input>  Name of csv file to download, must be within the /data directory.
<output>  Name of file to be saved, to be stored in /data directory. Include .csv filetype. 
" -> doc

library(tidyverse)
library(testthat)
library(docopt)

opt <- docopt(doc)

main <- function(input, output){
    df <- read_csv(paste('./data/', input, sep=""), skip=1)
    df <- df %>% select(-'0')
    df <- df %>% rename(
        DEFAULT_NEXT_MONTH= 'default payment next month',
        PAY_1= 'PAY_0'
    )
    write_csv(df, paste('./data/', output, sep=""))
}

test_main <- function(output){
  test_that("test main func should create csv", {
    df <- read_csv(paste('./data/', output, sep=""), skip=1)
        expect_equal(nrow(df), 29999)
  })
}

main(opt$input, opt$output)

test_main(opt$output)